packages/rulesets/src/spectral/az-arm.ts

```diff
@@ -158,6 +159,22 @@ const ruleset: any = {
       },
     },
 
+    // RPC Code: RPC-Async-V1-06
+    LroAzureAsyncOperationHeader: {
+      rpcGuidelineCode: "RPC-Async-V1-06",
+      description: "All long-running operations must include an `Azure-AsyncOperation' response header.",
+      message: "{{description}}",
+      severity: "error",
+      formats: [oas2],
+      given: ["$[paths,'x-ms-paths'].*.*[?(@property === 'x-ms-long-running-operation' && @ === true)]^.responses.*.headers"],
```

Comment:
I believe this will only pass in the headers object if it exists. If it does not, it will not call the function. I think you would need to make the given part like this
```suggestion
      given: ["$[paths,'x-ms-paths'].*.*[?(@property === 'x-ms-long-running-operation' && @ === true)]^.responses.*.*"],
```

Reply:
*.* is also skipping the headers check, have to tweak some code changes in lro-azure-async-operation-header.ts file